waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -952,67 +962,85 @@ proc appendRLNProof*(rlnPeer: WakuRLNRelay, msg: var WakuMessage,
 proc addAll*(wakuRlnRelay: WakuRLNRelay, list: seq[IDCommitment]): RlnRelayResult[void] =
   # add members to the Merkle tree of the  `rlnInstance`
   ## Returns an error if it cannot add any member to the Merkle tree
-  for i in 0..list.len-1:
-    let member = list[i]
-    let memberAdded = wakuRlnRelay.insertMember(member)
-    if not memberAdded.isOk():
-      return err(memberAdded.error())
+  let membersAdded = wakuRlnRelay.insertMembers(0, list)
+  if not membersAdded.isOk():
+    return err("failed to add members to the Merkle tree")
   return ok()
 
-type GroupUpdateHandler* = proc(pubkey: Uint256, index: Uint256): RlnRelayResult[void] {.gcsafe.}
+type MembershipTuple* = tuple[index: MembershipIndex, idComm: IDCommitment]
```

Comment:
I'd say there is no need for this new type, you can use `RlnMembershipCredentials` https://github.com/status-im/nwaku/blob/bcc6c322879007a5f6084660524dc9aa5bcd2fde/waku/v2/protocol/waku_rln_relay/waku_rln_relay_types.nim#L82

Reply:
This includes the MembershipKeyPair, which we don't have access to (idKey) - https://github.com/status-im/nwaku/blob/bcc6c322879007a5f6084660524dc9aa5bcd2fde/waku/v2/protocol/waku_rln_relay/waku_rln_relay_types.nim#L51-L58

Reply:
Yes, it does, it is part of the `membershipKeyPair ` field which is of type `MembershipKeyPair`.

Reply:
we don't have the other users idKeys. Functionally, the uses of the types are different.